web/src/search/input/QueryInput.tsx

```diff
@@ -141,35 +158,36 @@ export class QueryInput extends React.Component<Props, State> {
                         }
 
                         // Used to know if a filter value, and not just a separate word, is being typed
-                        const filterAndValueBeforeCursor = lastFilterAndValueBeforeCursor(queryState)
+                        const filterAndValueBeforeCursor = validFilterAndValueBeforeCursor(queryState)
 
                         // If a filter value is being typed but selected filter does not use
                         // fuzzy-search suggestions, then return only static suggestions
                         if (
                             filterAndValueBeforeCursor &&
-                            !fuzzySearchFilters.includes(filterAndValueBeforeCursor.filter)
+                            !fuzzySearchFilters.includes(filterAndValueBeforeCursor.resolvedFilterType)
                         ) {
                             return [{ suggestions: staticSuggestions }]
                         }
 
+                        // Because of API limitations, we need to modify the query before the request,
+                        // see definition of `formatQueryForFuzzySearch`
                         const queryForFuzzySearch = formatQueryForFuzzySearch(queryState)
                         const fullQuery = this.props.prependQueryForSuggestions
                             ? this.props.prependQueryForSuggestions + ' ' + queryForFuzzySearch
                             : queryForFuzzySearch
 
-                        // Dos a fuzzy search and formats suggestions for display.
-                        return fetchSuggestions(fullQuery).pipe(
+                        const fuzzySearch$ = fetchSuggestions(fullQuery).pipe(
```

Comment:
For consistency - our codebase doesn't use the dollar postfix for Observable variable names.

```suggestion
                        const fuzzySearchSuggestions = fetchSuggestions(fullQuery).pipe(
```